docs/user/tpm-authentication.md

```diff
@@ -0,0 +1,237 @@
+
+# Download Infineon TPM Intermediate CA  
+curl -O https://www.infineon.com/dgdl/Infineon-TPM_RSA_Intermediate_CA_01-C-v01_00-EN.cer
+
+# Convert to PEM format if needed
+openssl x509 -inform DER -in Infineon-TPM_RSA_Root_CA-C-v01_00-EN.cer \
+    -out infineon-root-ca.pem
+
+openssl x509 -inform DER -in Infineon-TPM_RSA_Intermediate_CA_01-C-v01_00-EN.cer \
+    -out infineon-intermediate-ca.pem
+```
+
+### Kubernetes Deployment
+
+For Kubernetes deployments, add TPM CA certificates via ConfigMap and configure the API server:
+
+1. **Create ConfigMap with TPM CA certificates:**
+
+```bash
+kubectl create configmap tpm-ca-certs \
+  --from-file=infineon-root-ca.pem \
+  --from-file=infineon-intermediate-ca.pem \
+  --namespace=flightctl
+```
+
+1. **Update the API server configuration to include TPM CA paths:**
+
+```yaml
+# Get current API configuration
+kubectl get configmap flightctl-api-config -n flightctl -o yaml
+
+# Add tpmCAPaths to the service section
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: flightctl-api-config
+  namespace: flightctl
+data:
+  config.yaml: |
+    service:
+      tpmCAPaths:
+        - /etc/flightctl/tpm-cas/infineon-root-ca.pem
+        - /etc/flightctl/tpm-cas/infineon-intermediate-ca.pem
+      # ... rest of service configuration
+```
+
+1. **Mount the TPM CA certificates in the API server deployment:**
+
+```bash
+kubectl patch deployment flightctl-api -n flightctl --type='json' -p='[
+  {
+    "op": "add",
+    "path": "/spec/template/spec/volumes/-",
+    "value": {
+      "name": "tpm-ca-certs",
+      "configMap": {
+        "name": "tpm-ca-certs"
+      }
+    }
+  },
+  {
+    "op": "add",
+    "path": "/spec/template/spec/containers/0/volumeMounts/-",
+    "value": {
+      "name": "tpm-ca-certs",
+      "mountPath": "/etc/flightctl/tpm-cas",
+      "readOnly": true
+    }
+  }
+]'
+```
+
+1. **Wait for the deployment to roll out:**
+
+```bash
+kubectl rollout status deployment/flightctl-api -n flightctl
+```
+
+### Standalone Deployment
+
+For standalone installations:
+
+```bash
+# Create TPM CA certificate directory
+sudo mkdir -p /etc/flightctl/tpm-cas
+
+# Copy TPM manufacturer CA certificates
+sudo cp infineon-root-ca.pem /etc/flightctl/tpm-cas/
+sudo cp infineon-intermediate-ca.pem /etc/flightctl/tpm-cas/
+
+# Set appropriate permissions
+sudo chmod 644 /etc/flightctl/tpm-cas/*.pem
+```
+
+Configure the API server:
+
+```yaml
+# /etc/flightctl/server.yaml or config.yaml
+service:
+  tpmCAPaths:
+    - /etc/flightctl/tpm-cas/infineon-root-ca.pem
+    - /etc/flightctl/tpm-cas/infineon-intermediate-ca.pem
+  # ... rest of service configuration
+```
+
+> [!NOTE]
+> Different TPM manufacturers provide their CA certificates in various locations:
+>
+> - **Infineon**: <https://www.infineon.com/cms/en/product/security-smart-card-solutions/optiga-embedded-security-solutions/>
+> - **STMicroelectronics**: Contact STMicroelectronics support for CA certificates
```

Comment:
https://www.st.com/resource/en/technical_note/tn1330-st-trusted-platform-module-tpm-endorsement-key-ek-certificates-stmicroelectronics.pdf has a bunch of certs in it